### PR TITLE
Alzheimer's Diagnosis Using  Visual Transformer s4698290

### PR DESCRIPTION
**Pull request by Marc519**
As per report project 6, this implements a visual transformer network based on the paper "An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale". The model achieves 67% on the test set despite extensive hyperparameter tuning, as well as achieving almost 80% on the validation set.

It also has code for a experimental convolutional transformer that was not used in the report due to bad overfitting to the validation set.

- `train.py`:  Trains the model using the hyperparameters to reproduce the results in README.md
- `dataset.py`:  Converts the dataset to a pytorch dataset, for use in a dataloader for training and testing
- `modules.py`: Implements the visual transformer model 
- `predict.py`:  Uses the model outputted by train.py to predict the labels of the test set and prints the accuracy
-  `README.md`: Contains a report detailing the results, model structure, interpretations, and information for reproducing results
-  `readmeImages/`: Contains the images used in the README file

The performance could likely be improved by tuning the experimental convolutional transformer more and adding learning rate scheduling.

The commits not done from this username are because they were done on rangpur

Author: Marc Mennes, s4698290

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: 
- Excellent and detailed


Commit messages: Excellent

Code:
- Uses Vision transformer
- Well commented
- Good design

Functionality/Performance:
- Did patient-level split
- Test acc is 67%
- It's a little bit weird that training accuracy is below val acc during training


General comments:
- Solves the problem well although performance is not over 80%
- Need a reference section

**Comment by Marc519** [comment]
Can i add a reference section now or is it too late?

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good, but needs refs -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
